File: Travel Package Management Java/src/com/example/travel/Activity.java
package com.example.travel;

import java.util.HashSet;
import java.util.Set;

public class Activity {
    private String name;
    private String description;
    private int cost;
    private int capacity;
    private String assignedDestination;
    private boolean assigned = false;
    private Set<Passenger> signedUpPassengers = new HashSet<>();

    public Activity(String name, String description, int cost, int capacity) {
        this.name = name;
        this.description = description;
        this.cost = cost;
        this.capacity = capacity;
    }

    public void printDetails() {
        System.out.println("Activity name: " + name + ", Description: " + description + ", Cost: " + cost + ", Capacity: " + capacity);
    }

    public double getCost() {
        return cost;
    }

    public String getName() {
        return name;
    }

    public void assignToDestination(String destination) {
        this.assignedDestination = destination;
        this.assigned = true;
    }

    public boolean isAssignedToDestination() {
        return assigned;
    }

    public boolean signUp(Passenger passenger) {
        if (signedUpPassengers.size() >= capacity) return false;
        return signedUpPassengers.add(passenger);
    }

    public boolean isSignedUp(Passenger passenger) {
        return signedUpPassengers.contains(passenger);
    }

    public boolean isAvailable() {
        return signedUpPassengers.size() < capacity;
    }

    public void printAvailableActivityDetails() {
        System.out.println("Activity Name: " + name + ", Spaces Available: " + (capacity - signedUpPassengers.size()));
    }
}

File: Travel Package management C++/Activity.cpp
#include "Activity.h"
#include "Passenger.h" 
#include <iostream>

using namespace std;


Activity::Activity(string n, string desc, int c, int cap){
    name = n;
    description = desc;
    cost = c;
    capacity = cap;
}

void Activity::printDetails(){
    cout << "Activity name: " << name << ", Description: " << description << ", Cost: " << cost << ", Capacity: " << capacity << endl;
}
double Activity::getCost(){ return cost; }
string Activity::getName(){ return name; }

void Activity::assignToDestination(string destination){
    assignedDestination = destination;
    assigned = true;
}

bool Activity::isAssignedToDestination(){
    return assigned;
}

bool Activity::signUp(shared_ptr<Passenger> passenger){
    if(signedUpPassengers.size() >= capacity)return false;
    return signedUpPassengers.insert(passenger).second;
}

bool Activity::isSignedUp(shared_ptr<Passenger> passenger){
    return signedUpPassengers.find(passenger) != signedUpPassengers.end();
}

bool Activity::isAvailable(){ return signedUpPassengers.size() < capacity; }

void Activity::printAvailableActivityDetails(){
    cout << "Activity Name: " << name << ", Spaces Available: " << (capacity - signedUpPassengers.size()) << endl;
}

